src/Features/ERDDAP/Platform/Observations/CurrentConditions/wind.tsx

```diff
@@ -1,81 +1,178 @@
 
 import { round } from "Shared/math"
 import { DataTimeSeries } from "Shared/timeSeries"
 import { compassDirection } from "Shared/unitConversion"
+import { useUnitSystem } from "Features/Units"
 import { UnitSystem } from "Features/Units/types"
 import { converter } from "Features/Units/Converter"
 
 import { WindTimeSeriesChart } from "components/Charts"
 
-import { PlatformDataset } from "../../../types"
+import { useDatasets } from "../../../hooks"
+import { PlatformFeature, PlatformTimeSeries } from "../../../types"
+import { pickWindDatasets, pickWindTimeSeries } from "../../../utils/wind"
+import { cardProps, observationLink } from "./common_card"
 
-interface Props {
-  datasets: PlatformDataset[]
-  unit_system: UnitSystem
+interface WindCardProps {
+  platform: PlatformFeature
 }
 
-export const WindCard: React.SFC<Props> = ({ datasets, unit_system }) => {
-  if (datasets.length < 1) {
-    return null
+/**
+ * If there are wind time series for a platform load and display recent data.
+ *
+ * @param platform Platform GeoJSON feature to display wind data from
+ */
+export const WindCard: React.FunctionComponent<WindCardProps> = ({ platform }) => {
+  const aDayAgo = new Date()
+  aDayAgo.setDate(aDayAgo.getDate() - 1)
+
+  const { windTimeSeries, timeSeries } = pickWindTimeSeries(platform, aDayAgo)
+
+  // No current data or wind data don't exist for the platform
+  if (windTimeSeries.length < 1) {
+    const noWindTimeSeries = platform.properties.readings.filter((timeSeries) =>
+      timeSeries.data_type.standard_name.includes("wind")
+    )
+
+    if (noWindTimeSeries.length < 1) {
+      return null
+    }
+
+    return <OtherWindCard platform={platform}>No recent wind data available.</OtherWindCard>
+  }
+
+  if (timeSeries.length < 0) {
+    return <OtherWindCard platform={platform}>No wind data</OtherWindCard>
+  }
+
+  return <LoadWindCard platform={platform} timeSeries={timeSeries} />
+}
+
+interface LoadWindCardProps extends WindCardProps {
+  timeSeries: PlatformTimeSeries[]
+}
+
+/**
+ * Load data for wind time series, and connect currently selected unit_system
+ */
+const LoadWindCard: React.FunctionComponent<LoadWindCardProps> = ({ platform, timeSeries }) => {
+  const unit_system = useUnitSystem()
+  const { isLoading, data } = useDatasets(timeSeries)
+
+  if (isLoading) {
+    return <OtherWindCard platform={platform}>Loading wind data</OtherWindCard>
+  }
+
+  if (data) {
+    const aDayAgo = new Date()
+    aDayAgo.setDate(aDayAgo.getDate() - 1)
+
+    const filteredDatasets = (data as DataTimeSeries[])
+      .map((ds) => ({
+        ...ds,
+        timeSeries: ds.timeSeries.filter((r) => aDayAgo < r.time),
+      }))
+      .filter((ds) => ds.timeSeries.length > 0)
+
+    if (filteredDatasets.length > 0) {
+      return (
+        <DisplayWindCard
+          platform={platform}
+          timeSeries={timeSeries}
+          datasets={filteredDatasets}
+          unit_system={unit_system}
```

Comment:
Codacy found an issue: [Identifier 'unit_system' is not in camel case.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)